components/InputField.js

```diff
@@ -396,48 +398,38 @@ class InputField extends React.Component {
         );
         break;
 
-      case 'country':
+      case 'address':
         this.input = (
-          <div>
-            {horizontal && (
-              <Flex flexWrap="wrap" p={1}>
-                <Box width={[1, 2 / 12]}>
-                  <label>{capitalize(field.label)}</label>
-                </Box>
-                <Box width={[1, 10 / 12]}>
-                  <InputTypeCountry
-                    name={field.name}
-                    inputId={field.name}
-                    value={field.value}
-                    defaultValue={field.defaultValue}
-                    onChange={this.handleChange}
-                  />
-                </Box>
-              </Flex>
-            )}
-            {!horizontal && (
-              <Flex flexWrap="wrap" p={1}>
-                {field.label && (
-                  <Box width={1}>
-                    <label>{`${capitalize(field.label)}`}</label>
-                  </Box>
-                )}
-                <Box width={1}>
-                  <InputTypeCountry
-                    name={field.name}
-                    inputId={field.name}
-                    value={field.value}
-                    defaultValue={field.defaultValue}
-                    onChange={this.handleChange}
-                  />
-                  {field.description && <HelpBlock>{field.description}</HelpBlock>}
-                </Box>
-              </Flex>
+          <Flex flexWrap="wrap" p={1}>
+            {field.label && (
+              <Box width={1}>
+                <Label label={capitalize(field.label)} isPrivate={field.isPrivate} />
+              </Box>
             )}
-          </div>
+            <Box width={1}>
+              <StyledInputLocation
+                location={this.state.value || field.defaultValue}
+                onChange={event => this.handleChange(event)}
+              />
+              {field.description && <HelpBlock>{field.description}</HelpBlock>}
+            </Box>
+          </Flex>
+        );
+        break;
+      case 'socialLinks':
+        this.input = (
+          <Box p={1}>
+            <Box my="5px" fontWeight={700}>
+              <FormattedMessage defaultMessage="Social Links" />
+            </Box>
+            <SocialLinksFormField
+              value={this.state.value || field.defaultValue}
+              touched={this.state.modified}
+              onChange={event => this.handleChange(event)}
+            />
+          </Box>
         );
         break;
```

Comment:
I don't really get why we're adding social links here?

Reply:
Oops, fixed this by passing down the `state.modified` from Form.js as `modifiedForm` to InputField.js